index.bs

```diff
@@ -4087,6 +4170,65 @@ readableStrategy)">new TransformStream(<var>transformer</var> = {}, <var>writabl
   1. <a>Resolve</a> _startPromise_ with _startResult_.
 </emu-alg>
 
+<h4 id="transformer-api">Transformer API</h4>
+
+The {{TransformStream()}} constructor accepts as its first argument a JavaScript object representing the
+<a>transformer</a>. Such objects may contain any of the following methods:
+
+<dl>
+  <dt><dfn method for="transformer">start(<var>controller</var>)</dfn></dt>
+  <dd>
+    <p>A function that is called immediately upon creation of the {{TransformStream}}.</p>
+
+    <p>Typically this is used enqueue prefix <a>chunks</a>, using
+    {{TransformStreamDefaultController/enqueue()|controller.enqueue()}}. Those chunks will be read from the <a>readable
+    side</a> but don't depend on any writes to the <a>writable side</a>.</p>
+
+    <p>If this initial process is asynchronous, for example by virtue of taking some effort to acquire the prefix
+    chunks, the function can return a promise to signal success or failure; a rejected promise will error the stream.
+    Any thrown exceptions will be re-thrown by the {{TransformStream()}} constructor.</p>
+  </dd>
+
+  <dt><dfn method for="transformer">transform(<var>chunk</var>, <var>controller</var>)</dfn></dt>
+  <dd>
+    <p>A function called when a new <a>chunk</a> originally written to the <a>writable side</a> is ready to be
+    transformed. The stream implementation guarantees that this function will be called only after previous transforms
+    have succeeded, and never before {{underlying sink/start()}} has completed or after {{transformer/flush()}} has been
+    called.</p>
+
+    <p>This function performs the actual transformation work of the transform stream. It can enqueue the results using
+    {{TransformStreamDefaultController/enqueue()|controller.enqueue()}}. This permits a single chunk written to the
+    writable side to result in zero or multiple chunks on the <a>readable side</a>, depending on how many times
+    {{TransformStreamDefaultController/enqueue()|controller.enqueue()}} is called. [[#example-ts-lipfuzz]] demonstrates
+    this by sometimes enqueuing zero chunks.</p>
+
+    <p>If the process of transforming is asynchronous, this function can return a promise to signal success or failure
+    of the transformation. A rejected promise will error both the readable and writable sides of the transform
+    stream.</p>
+
+    <p>If no {{transformer/transform()}} is supplied, the identity transform is used, which enqueues chunks unchanged
+    from the writable side to the readable side.</p>
+  </dd>
+
+  <dt><dfn method for="transformer">flush(<var>controller</var>)</dfn></dt>
+  <dd>
+    <p>A function called after all <a>chunks</a> written to the <a>writable side</a> have been transformed by
+    successfully passing through {{transformer/transform()}}, and the writable side is about to be closed.</p>
+
+    <p>Typically this is used to enqueue suffix chunks to the <a>readable side</a>, before that too becomes closed. An
```

Comment:
Do you think we should mention that there's no need to explicitly call `controller.close()` inside `flush()`? I made this mistake in my early text encoder stream implementation.

Reply:
Sure, seems reasonable now that we have all this extra space.